### ISSUE TITLE
Debug in R markdown file

### ISSUE THREAD

**Issue by taooceros** [ManuelHentschel/VSCode-R-Debugger#153]
Is it possible to debug for a code chunk within a r markdown file? I think python jupyter notebook does support the feature, same as r studio.

**Comment by markusgumbel** [comment]
I wonder if you could consider a chunk just a piece of code to be executed in the global environment. What variables are defined at debugging time is up to the user. He/she could run any chunks above (that would create those global variables) before debugging whenever needed. 

Debugging a chunk is a very important feature in my opinion. Jetbrains' R plug in can do so - RStudio not.

But anyway, VS code's R features incl. this debugger are already excellent! Please  go on!

**Comment by JZL** [comment]
I agree and for what it's worth, where I find using the `browser()` debugger useful in rstudio .rmd files is to debug external libraries/other R files, stepping through the function calls to see where an error/bug is coming from. So I wouldn't need/want anything very crazy (or even cross rmd cell boundaries) -- just pulling out the current cell into an execution environment along with all the global/local vars.

If I made the plunge to use VScode over Rstudio I could definitely use `purl`ing to get it to work but I wanted to be careful about not losing the current environment.

**Comment by milanglacier** [comment]
> > Start a debug session with an empty workspace and send the code inside a chunk to the debug console (e.g. using ctrl+enter or the buttons provided by the vscode-R extension). To an extent this might already work.
> 
> That looks reasonable, and I remember the jupyter notebook do the same thing. Convert the whole document to r will lose the advantage of rmd. Though I think the second option may be easier.

`purl()` exported `.R` file will have your code chunk separated by `## -----------------`, 

and you just simple replace `## -----------------` by `# %%`, and you can use the command palette `run selected chunk `, which makes a `R` file runs blockwisely like `rmd`
